src/dev/eslint/run_eslint_custom.ts

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import path from 'path';
+import os from 'os';
+import execa from 'execa';
+import { run } from '@kbn/dev-cli-runner';
+import { REPO_ROOT } from '@kbn/repo-info';
+import type { Package } from '@kbn/repo-packages';
+import { getPackages } from '@kbn/repo-packages';
+import type { ToolingLog } from '@kbn/tooling-log';
+
+interface EslintJsonMessage {
+  ruleId?: string;
+  message?: string;
+}
+
+interface EslintJsonFile {
+  filePath: string;
+  messages: EslintJsonMessage[];
+}
+
+interface EslintRunResult {
+  name: string;
+  directory: string;
+  report: EslintJsonFile[];
+  exitCode: number;
+  stderr?: string;
+}
+
+type FormatterFn = (log: ToolingLog, results: EslintRunResult[]) => void;
+
+run(
+  async ({ log, flags }) => {
+    const eslintConf = String(flags['eslint-config']);
+    const formatter = String(flags.formatter);
+    const concurrency = Number(flags.concurrency);
+    const filter = typeof flags.filter === 'string' ? (flags.filter as string) : '';
+
+    const availableParallelism = os.availableParallelism() ?? 1;
+
+    if (concurrency > availableParallelism) {
+      log.warning(
+        `Available parallelism (${availableParallelism}) is lower than requested concurrency (${concurrency}). Will use maximum allowed instead (${availableParallelism}).`
+      );
+    }
+
+    const usedConcurrency = Math.max(
```

Comment:
this is two of the same done in one place :) either Math.max or the ternary would do